modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RapidExperimentationTest.java

```diff
@@ -0,0 +1,328 @@
+ *******************************************************************************/
+public class RapidExperimentationTest extends TestBase {
+    /**
+     * This test class contains functional tests that check the rapid experimentation functionality
+     * 
+     * Created on 05/23/2017
+     */
+    private static final int RAPID_EXP_MAX_USERS = 5; // max cap on number of users
+    private static final int NUMBER_OF_EXPERIMENTS_PER_BATCH = 5;
+    private static final int NUMBER_OF_RAPID_EXPERIMENTS_IN_BATCH = 2; // number of rapid experiments in a batch that
+                                                                       // has combination of normal and rapid
+                                                                       // experiments
+
+    private static final String BATCH_PAGE = "rapidexperiment_page";
+    private static final String PARTIAL_BATCH_PAGE = "rapidexperiment_partialpage";
+    Experiment rapidExperiment1 = null;
+    Experiment rapidExperiment2 = null;
+    Experiment rapidExperiment3 = null;
+    Page page = null;
+    List<Experiment> expList = new ArrayList<Experiment>();
+    List<Experiment> batchExperiments = new ArrayList<Experiment>();
+    List<Experiment> experimentsListOfRapidApplication = new ArrayList<Experiment>();
+
+    Application application = null;
+
+    @BeforeClass
+    public void setUp() {
+
+        application = new Application("rapidExperimentApplication");
+
+        experimentsListOfRapidApplication = getApplicationExperiments(application);
+        pauseTerminateAndDeleteExperiments(experimentsListOfRapidApplication);
+
+        rapidExperiment1 = ExperimentFactory.createExperiment("_1" + UUID.randomUUID().toString(), -1)
+                .setApplication(application).setIsRapidExperiment(true).setUserCap(RAPID_EXP_MAX_USERS);
+        rapidExperiment2 = ExperimentFactory.createExperiment("_2" + UUID.randomUUID().toString(), -1)
+                .setApplication(application);
+
+        rapidExperiment3 = ExperimentFactory.createExperiment("_3" + UUID.randomUUID().toString(), -1)
+                .setApplication(application).setIsRapidExperiment(true).setUserCap(RAPID_EXP_MAX_USERS);
+
+        rapidExperiment1 = postExperiment(rapidExperiment1);
+        rapidExperiment2 = postExperiment(rapidExperiment2);
+        rapidExperiment3 = postExperiment(rapidExperiment3);
+        expList.add(rapidExperiment1);
+        expList.add(rapidExperiment2);
+        expList.add(rapidExperiment3);
+        for (int i = 1; i <= 3; i++) {
+            Experiment exp = expList.get(i - 1);
+            List<Bucket> bucketList = BucketFactory.createBuckets(exp, 3);
+            postBuckets(bucketList);
+
+            exp.state = Constants.EXPERIMENT_STATE_RUNNING;
+            exp = putExperiment(exp);
+
+        }
+
+    }
+
+    @Test
+    public void testRapidExperiment() {
+
+        Experiment exp = getExperiment(expList.get(0));
```

Comment:
Good suggestion... i will make each test more atomic....so that experiment creation will be part of the test...